### PR TITLE
Contact mail

### PR DESCRIPTION
**Pull request by emreesendir**
- New firebase function was added to receive contact form data and send emails.
- mail.ts file created for email related firebase functions
- contacts page edited
- workflow configuration files edited

### CODE REVIEW

File: .github/workflows/deploy-firebase-acc.yml [hunk 1]
```diff
@@ -18,14 +18,16 @@ jobs:
       - name: Configure
         uses: w9jds/firebase-action@master
         with:
-          args: functions:config:set clothingloop.base_domain="$BASE_DOMAIN" clothingloop.region="$REGION" clothingloop.admin_emails="$ADMIN_EMAILS"
+          args: functions:config:set clothingloop.base_domain="$BASE_DOMAIN" clothingloop.region="$REGION" clothingloop.admin_emails="$ADMIN_EMAILS" contactmail.to="$CONTACT_MAIL_TO" contactmail.cc="$CONTACT_MAIL_CC"
```
Comment (timstokman) [inline .github/workflows/deploy-firebase-acc.yml:21]: Best to stick to the same naming conventions for configuration.

`contactmail.to` vs `clothingloop.admin_emails` and `clothingloop.base_domain`.

If you don't agree with a convention, refactor it all, don't mix things.

File: firebase/functions/src/index.ts [hunk 2]
```diff
@@ -277,3 +277,50 @@ export const getUserByEmail =
...
+        });
+
+        // send confirmation mail to the user
+        await db.collection("mail").add({
+          to: email,
+          message: {
+            subject: "We Recieved Your Message",
+            html: ` <h1>Thank you ${name} for your message!</h1>
+                    <p>We recieved your message. Somebody will contact you soon.</p>
+                    <h5>Your Message</h5>
```
Comment (timstokman) [inline firebase/functions/src/index.ts:319]: I'd do:

`<h5>Your Message:</h5>`

File: .github/workflows/deploy-firebase-acc.yml [hunk 3]
```diff
@@ -18,14 +18,16 @@ jobs:
...
         env:
           ADMIN_EMAILS: ${{ secrets.ADMIN_EMAILS }}
           BASE_DOMAIN: ${{ secrets.BASE_DOMAIN }}
           FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
           PROJECT_ID: ${{ secrets.FIREBASE_PROJECT_ID }}
           PROJECT_PATH: ./firebase/functions
           REGION: ${{ secrets.REGION }}
+          CONTACT_MAIL_TO: ${{ secrets.CONTACT_MAIL_TO }}
+          CONTACT_MAIL_CC: ${{ secrets.CONTACT_MAIL_CC }}
```
Comment (timstokman) [inline .github/workflows/deploy-firebase-acc.yml:30]: Consider making this a single setting. "to" can already be an array: https://medium.com/firebase-developers/firebase-extension-trigger-email-5802800bb9ea

File: .github/workflows/deploy-firebase-prod.yml [hunk 4]
```diff
@@ -18,14 +18,16 @@ jobs:
...
+          args: functions:config:set clothingloop.base_domain="$BASE_DOMAIN" clothingloop.region="$REGION" clothingloop.admin_emails="$ADMIN_EMAILS" contactmail.to="$CONTACT_MAIL_TO" contactmail.cc="$CONTACT_MAIL_CC"
         env:
           ADMIN_EMAILS: ${{ secrets.ADMIN_EMAILS }}
           BASE_DOMAIN: ${{ secrets.BASE_DOMAIN }}
           FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
           PROJECT_ID: ${{ secrets.FIREBASE_PROJECT_ID }}
           PROJECT_PATH: ./firebase/functions
           REGION: ${{ secrets.REGION }}
+          CONTACT_MAIL_TO: ${{ secrets.CONTACT_MAIL_TO }}
+          CONTACT_MAIL_CC: ${{ secrets.CONTACT_MAIL_CC }}
```
Comment (timstokman) [inline .github/workflows/deploy-firebase-prod.yml:30]: Consider making this a single setting? Trigger mail to can already be an array: https://medium.com/firebase-developers/firebase-extension-trigger-email-5802800bb9ea

File: firebase/functions/src/index.ts [hunk 5]
```diff
@@ -277,3 +277,50 @@ export const getUserByEmail =
...
+                    <p>${message}</p>
+            `,
+          },
+        });
+
+        // send confirmation mail to the user
+        await db.collection("mail").add({
+          to: email,
+          message: {
+            subject: "We Recieved Your Message",
```
Comment (timstokman) [inline firebase/functions/src/index.ts:316]: "Recieved" is misspelled.

I'd personally make it:

subject: `"Thank you for contacting Clothing-Loop"`

html:
```
`<p>Hi ${name},</p>` +
`<p>Thank you for your message! You wrote:</p><p>` +
message +
"</p><p>We will contact you as soon as possible.</p><p>Regards,</p><p>The clothing-loop team!</p>";
```

Comment (emreesendir) [reply firebase/functions/src/index.ts]: Thank you for your feedback. I tried to apply them all.

**Review by timstokman — APPROVED** [review]
Good work on the PR!

**Pull request merged**